ycms/cms/views/ward/ward_view.py

```diff
@@ -0,0 +1,74 @@
+from django.shortcuts import redirect
+from django.views.generic.detail import DetailView
+
+from ...models import Room, Ward
+
+
+class WardView(DetailView):
+    """
+    View to see a ward
+    """
+
+    model = Ward
+    template_name = "ward/ward.html"
+    context_object_name = "ward"
+
+    def get_context_data(self, **kwargs):
+        """
+        This function returns a list of all rooms in the ward
+
+        :param kwargs: The supplied keyword arguments
+        :type kwargs: dict
+
+        :return: Response for filtered offers
+        :rtype: ~django.template.response.TemplateResponse
+        """
+
+        context = super().get_context_data(**kwargs)
+        ward = Ward.objects.get(id=self.kwargs["pk"])
+        ward_id = self.kwargs["pk"]
+        rooms = Room.objects.filter(ward_id=ward_id)
+        wards = Ward.objects.all()
+        context["rooms"] = rooms
+        context["corridor_index"] = str(len(rooms) // 2)
+        context["ward"] = ward
+        context["room_genders"] = self._get_room_genders(rooms)
+        context["patient_info"] = self._get_patient_info(ward.patients)
+        context["wards"] = wards
+        context["selected_ward_id"] = ward_id
+        return context
```

Comment:
```suggestion
    def get_context_data(self, **kwargs):
        """
        This function returns a list of all rooms in the ward

        :param kwargs: The supplied keyword arguments
        :type kwargs: dict

        :return: Response for filtered offers
        :rtype: ~django.template.response.TemplateResponse
        """
        ward_id = self.kwargs.get("pk")
        ward = Ward.objects.get(id=ward_id)
        rooms = ward.rooms.all()
        wards = Ward.objects.all()
        return {
            "rooms": rooms,
            "corridor_index": str(len(rooms) // 2),
            "ward": ward,
            "room_genders": self._get_room_genders(rooms),
            "patient_info": self._get_patient_info(ward.patients),
            "wards": wards,
            "selected_ward_id": ward_id,
            **super().get_context_data(**kwargs),
        }
```
Just a bit of simplification

Reply:
done

Reply:
Just a tip, you can click the "Commit suggestion" button instead of having to do it by hand 😄 
If you do need to do something by hand, jus click the "Resolve conversation" button afterwards!